dask/config.py

```diff
@@ -147,22 +147,30 @@ def collect_yaml(paths=paths):
     for path in paths:
         if os.path.exists(path):
             if os.path.isdir(path):
-                file_paths.extend(sorted([
-                    os.path.join(path, p)
-                    for p in os.listdir(path)
-                    if os.path.splitext(p)[1].lower() in ('.json', '.yaml', '.yml')
-                ]))
+                try:
+                    file_paths.extend(sorted([
+                        os.path.join(path, p)
+                        for p in os.listdir(path)
+                        if os.path.splitext(p)[1].lower() in ('.json', '.yaml', '.yml')
+                    ]))
+                except OSError:
+                    # Ignore permission errors
```

Comment:
Should we lightly warn here or no?

Reply:
I thought about that, but have mixed opinions on that. Could go either way, went with no warning for now because:

- We could only warn if the directory containing the nonreadable file was readable, otherwise we'd never attempt and would never warn (e.g. we could warn if `/etc` was readable but not `/etc/dask`, if `/etc` was unreadable we'd never get the chance to warn.
- Other tools like git also silently ignore unreadable config files.

Reply:
OK.  Fine by me.